### PR TITLE
[Currency] Add isBase flag on CurrencyInterface

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q                    | A
| -------------       | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | 
| License       | MIT

Added all methods required to set a Currency as a base one. This PR is connected with #3814, because now during the installation we get only one currency from the user, and we want to set it as the base one.

### CODE REVIEW

**Comment by pamil** [comment]
Somehow I find `isBase`counterintuitive, is there any reason why wouldn't we go for `isDefault`?
Also, we need some way to ensure that there is only one currency with that field enabled.

**Comment by pjedrzejewski** [comment]
@pamil @TheMadeleine DefaultableInterface :D

File: src/Sylius/Bundle/CurrencyBundle/Resources/config/doctrine/model/Currency.mongodb.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
         <field fieldName="code" name="code" type="string" length="3" />
         <field fieldName="exchangeRate" name="exchangeRate" type="float" />
         <field fieldName="enabled" name="enabled" type="boolean" />
+        <field name="isBase" column="is-base" type="boolean" />
```
Comment (stloyd) [inline src/Sylius/Bundle/CurrencyBundle/Resources/config/doctrine/model/Currency.mongodb.xml]: Hmmm, every other column is called in camelCase, why changing approach for this one?

File: src/Sylius/Component/Currency/spec/Model/CurrencySpec.php [hunk 2]
```diff
@@ -91,6 +91,17 @@ function it_can_set_enabled_value()
...
 
+    function it_is_not_base_currency_by_default()
+    {
+        $this->isBase()->shouldReturn(false);
+    }
+
+    function it_can_can_be_base_currency()
+    {
+        $this->setBase(true);
+        $this->isBase()->shouldReturn(true);
```
Comment (stloyd) [inline src/Sylius/Component/Currency/spec/Model/CurrencySpec.php]:
```php
$this->shouldBeBase();

File: src/Sylius/Component/Currency/spec/Model/CurrencySpec.php [hunk 3]
```diff
@@ -91,6 +91,17 @@ function it_can_set_enabled_value()
         $this->shouldNotBeEnabled();
     }
 
+    function it_is_not_base_currency_by_default()
+    {
+        $this->isBase()->shouldReturn(false);
```
Comment (stloyd) [inline src/Sylius/Component/Currency/spec/Model/CurrencySpec.php]:
```php
$this->shouldNotBeBase();

File: src/Sylius/Component/Currency/spec/Model/CurrencySpec.php (hunk 2 above)
Comment (pamil) [inline src/Sylius/Component/Currency/spec/Model/CurrencySpec.php]: IMHO the current code is more verbose and easier to understand than `shouldBeBase()`.

**Comment by pjedrzejewski** [comment]
Thanks Magda!

**Pull request merged**